File: eukay/lib/pages/profile/mappers/address_model.dart
class AddressModel {
  final String id, fullName, province, municipality, barangay, street, contact;
  final bool inUse;

  AddressModel({
    required this.id,
    required this.fullName,
    required this.province,
    required this.municipality,
    required this.barangay,
    required this.street,
    required this.contact,
    required this.inUse,
  });

  factory AddressModel.fromJson(Map<String, dynamic> json) {
    return AddressModel(
      id: json['_id'],
      fullName: json['fullName'],
      province: json['province'],
      municipality: json['municipality'],
      barangay: json['barangay'],
      street: json['street'],
      contact: json['contact'],
      inUse: json['inUse'],
    );
  }
}

File: server/models/address.model.js
const mongoose = require("mongoose");

const AddressSchema = mongoose.Schema(
  {
    fullName: {
      type: String,
      required: true,
    },

    contact: {
      type: String,
      required: true,
    },

    province: {
      type: String,
      required: false,
      default: "Pangasinan",
    },

    municipality: {
      type: String,
      required: true,
    },

    barangay: {
      type: String,
      required: true,
    },

    street: {
      type: String,
      required: true,
    },

    inUse: {
      type: Boolean,
      default: false,
    },

    userId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
      required: true,
    },
  },
  {
    timestamps: true,
  }
);
const Address = mongoose.model("Address", AddressSchema);

module.exports = Address;
